File: lib/models/bus_schedule.dart

import 'package:freezed_annotation/freezed_annotation.dart';

import 'bus_model.dart';
import 'bus_route.dart';
part 'bus_schedule.freezed.dart';
part 'bus_schedule.g.dart';
@unfreezed
class BusSchedule with _$BusSchedule{
  factory BusSchedule({
    int? scheduleId,
    required Bus bus,
    required BusRoute busRoute,
    required String departureTime,
    required int ticketPrice,
    @Default(0)int discount,
    @Default(100)int processingFee,
  }) = _BusSchedule;

  factory BusSchedule.fromJson(Map<String, dynamic> json) =>
      _$BusScheduleFromJson(json);

}

File: backend/ReservationBus/src/main/java/com/busBooking/ReservationBus/entity/BusSchedule.java
package com.busBooking.ReservationBus.entity;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "bus_schedule")
public class BusSchedule {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long scheduleId;
    
    @OneToOne
    @JoinColumn(name = "bus_id")
    private Bus bus;

    @OneToOne
    @JoinColumn(name = "bus_route_id")
    private BusRoute busRoute;

    private String departureTime;
    private Integer ticketPrice;
    private Integer discount;
    private Integer processingFee;
}
